demos/python_demos/multi_camera_multi_person_tracking/utils/network_wrappers.py

```diff
@@ -81,3 +105,143 @@ def forward(self, batch):
+        im_data = np.pad(im_data, ((0, 0),
+                                   (0, self.h - im_data.shape[1]),
+                                   (0, self.w - im_data.shape[2])),
+                         mode='constant', constant_values=0).reshape(1, self.c, self.h, self.w)
+        im_info = im_info.reshape(1, *im_info.shape)
+        output = self.net.net.infer(dict(im_data=im_data, im_info=im_info))
+
+        classes = output['classes']
+        valid_detections_mask = classes > 0
+        classes = classes[valid_detections_mask]
+        boxes = output['boxes'][valid_detections_mask]
+        scores = output['scores'][valid_detections_mask]
+        masks = output['raw_masks'][valid_detections_mask]
+        return boxes, classes, scores, np.full(len(classes), 0, dtype=np.int32), masks
+
+    def get_detections(self, frames, return_cropped_masks=True, only_class_person=True):
+        outputs = []
+        for frame in frames:
+            data_batch = self.preprocess(frame)
+            im_data = data_batch['im_data']
+            im_info = data_batch['im_info']
+            meta = data_batch['meta']
+
+            boxes, classes, scores, _, masks = self.forward(im_data, im_info)
+            scores, classes, boxes, masks = postprocess(scores, classes, boxes, masks,
+                                                        im_h=meta['original_size'][0],
+                                                        im_w=meta['original_size'][1],
+                                                        im_scale_y=meta['processed_size'][0] / meta['original_size'][0],
+                                                        im_scale_x=meta['processed_size'][1] / meta['original_size'][1],
+                                                        full_image_masks=True, encode_masks=False,
+                                                        confidence_threshold=self.confidence)
+            frame_output = []
+            for i in range(len(scores)):
+                if only_class_person and classes[i] != 1:
+                    continue
+                bbox = [int(value) for value in boxes[i]]
+                if return_cropped_masks:
+                    left, top, right, bottom = bbox
+                    mask = masks[i][top:bottom, left:right]
+                else:
+                    mask = masks[i]
+                frame_output.append([bbox, scores[i], mask])
+            outputs.append(frame_output)
+        return outputs
+
+    def run_async(self, frames, index):
+        self.frames = frames
+
+    def wait_and_grab(self):
+        return self.get_detections(self.frames)
+
+    def resize(self, sample):
+        image_height, image_width = sample['image'].shape[:2]
+        scale = min(self.h / image_height, self.w / image_width)
+        sample['image'] = cv2.resize(sample['image'], None, fx=scale, fy=scale)
+        sample['image'] = sample['image'].astype('float32').transpose(2, 0, 1)
```

Comment:
Type casting and transposition doesn't belong to resizing. I suggest moving the line into `preprocess(self, frame)`.

Reply:
Moved code from `resize` to `preprocess`.